templates/container-instances.json

```diff
@@ -0,0 +1,98 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "containerName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the Azure Container Instances resource. This will be its identifier in Azure and can be different from the image name."
+      }
+    },
+    "containerGroupName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the container group to identify individual container to support multi-container deployment. This is an optional parameter"
+      },
+      "defaultValue": ""
+    },
+    "imageName": {
+      "type": "string",
+      "metadata": {
+        "description": "Container image reference this could be the URI or Docker Hub tag of the public or private image. For Docker Hub images, the full URI isn't required, only the image name and tag (for example: microsoft/iis:nanoserver)"
+      }
+    },
+    "osType": {
+      "type": "string",
+      "defaultValue": "Linux",
+      "allowedValues": [
+        "Linux",
+        "Windows"
+      ],
+      "metadata": {
+        "description": "Windows or Linux container instance"
+      }
+    },
+    "numberCpuCores": {
+      "defaultValue": "1",
+      "type": "string",
+      "metadata": {
+        "description": "The number of CPU cores to allocate to the container. Must be an integer."
+      }
+    },
+    "memory": {
+      "defaultValue": "1.5",
+      "type": "string",
+      "metadata": {
+        "description": "The amount of memory to allocate to the container in gigabytes."
+      }
+    },
+    "restartPolicy": {
+      "type": "string",
+      "defaultValue": "Always",
+      "allowedValues": [
+        "OnFailure",
+        "Always",
+        "Never"
+      ],
+      "metadata": {
+        "description": "The behavior of Azure runtime if container has stopped."
+      }
+    },
+    "environmentVariables": {
+      "type": "array",
+      "metadata": {
+        "description": "List of environment variables in an array format. Provide a name and value field for each variable and name and securevalue for hidden attributes i.e. secrets and password "
+      }
+    }
+  },
+  "variables": {
+    "containerGroupName": "[if(not(empty(parameters('containerGroupName'))), parameters('containerGroupName') , parameters('containerName'))]"
+  },
+  "resources": [
+    {
+      "location": "[resourceGroup().location]",
+      "name": "[parameters('containerName')]",
```

Comment:
```suggestion
      "name": "[parameters('containerGroupName')]",
```

Reply:
I have renamed it so now it is called ContainerName and ContainerInstanceName to avoid confusions